proposals/0031-hlsl-vector-matrix-operations.md

```diff
@@ -18,43 +18,44 @@
+vector/matrix operations described in [Proposal 0029].
 
-[0029]: 0029-cooperative-vector.md
+[Proposal 0029]: 0029-cooperative-vector.md
 
 ## Motivation
 
 Modern GPUs have dedicated silicon to accelerate matrix operations, but HLSL
 doesn't provide a mechanism to easily utilize these units. Evaluation of
 matrix-vector operations (multiply, muladd, accumulation) in HLSL was previously
 scalarized at the DXIL level making it hard to employ these specialized units.
-This proposal builds on the "Long vectors" feature described in [0026],
+This proposal builds on the "Long vectors" feature described in [Proposal 0026],
 providing a mechanism to express matrix-vector ops in HLSL that can be lowered
-to the DXIL ops described [0029], these primitives provide the right level of
-abstraction for hardware acceleration.
+to the DXIL ops described [Proposal 0029], these primitives provide the right
+level of abstraction for hardware acceleration.
 
 An HLSL API needs to be defined to expose these new operations in a way that:
-* works well with existing HLSL APIs
-* is expected to work well with future HLSL APIs in the same problem space
-* can be implemented reasonably in DXC and cleanly in clang
 
-[0026]: 0026-hlsl-long-vector-type.md
+- works well with existing HLSL APIs
+- is expected to work well with future HLSL APIs in the same problem space
+- can be implemented reasonably in DXC and cleanly in clang
+
+[Proposal 0026]: 0026-hlsl-long-vector-type.md
 
 ## Proposed solution
 
-This API will be implemented using HLSL code.  The exact mechanism for getting
+This API will be implemented using HLSL code. The exact mechanism for getting
 this code into a developer's shader is TBD, but implementations have a few
 possible options, including:
 
-* Developers must explicitly #include a header file
-* The compiler force-includes the header file
-* The compiler force-includes a precompiled version of the header file
+- Developers must explicitly #include a header file
+- The compiler force-includes the header file
+- The compiler force-includes a precompiled version of the header file
 
-The header-implementation accesses the DXIL operations described in [0029] by
-calling low-level builtins. These builtins should be considered implementation
-details and users should not call them directly. However, since they are a part
-of the implementation plan for the first implementation of this API they are
-described [below](#builtins).
+The header-implementation accesses the DXIL operations described in [Proposal
+0029] by calling low-level builtins. These builtins should be considered
+implementation details and users should not call them directly. However, since
+they are a part of the implementation plan for the first implementation of this
+API they are described [below](#builtins).
 
 Since this API is currently only supported by DirectX, all the new types /
 methods described in it are placed in the `dx` namespace. Within this namespace,
```

Comment:
Is dx a new namespace for hlsl or have we used it before?

Reply:
We're using it for SER as well.  

I don't think we shipped anything that uses it.  More background here:

https://github.com/microsoft/hlsl-specs/blob/main/docs/Process.md#fast-track-for-extensions